lib/msal-common/src/error/InteractionRequiredAuthError.ts

```diff
@@ -10,41 +10,48 @@ import { ServerError } from "./ServerError";
  * InteractionRequiredAuthErrorMessage class containing string constants used by error codes and messages.
  */
 export const InteractionRequiredAuthErrorMessage = {
-    interactionRequired: {
-        code: "interaction_required"
-    },
-    consentRequired: {
-        code: "consent_required"
-    },
-    loginRequired: {
-        code: "login_required"
-    }
+    interactionRequired: "interaction_required",
+    consentRequired: "consent_required",
+    loginRequired: "login_required"
+};
+
+export const InteractionRequiredAuthSubErrorMessage = {
+    messageOnly: "message_only",
+    additionalAction: "additional_action",
+    basicAction: "basic_action",
+    userPasswordExpired: "user_password_expired",
+    consentRequired: "consent_required"
 };
 
 /**
  * Error thrown when user interaction is required at the auth server.
  */
 export class InteractionRequiredAuthError extends ServerError {
+    subError: string;
 
-    constructor(errorCode: string, errorMessage?: string) {
+    constructor(errorCode: string, errorMessage?: string, subError?: string) {
         super(errorCode, errorMessage);
         this.name = "InteractionRequiredAuthError";
+        this.subError = InteractionRequiredAuthError.validateSubError(subError);
```

Comment:
Curious why validation is needed, are there suberrors that will be returned from the service that we don't want to support?

Reply:
This can be discussed with the team. I can't think of any we wouldn't want to surface. I was going off the previous research Jake had done [here](https://identitydivision.visualstudio.com/DevEx/_git/AuthLibrariesApiReview?path=%2FMSALJS_APIChanges%2FSubErrorCodes.md&version=GBsub-error-codes&_a=preview), thinking it had already been agreed upon. He mentioned there might be scenarios where we don't want to surface certain sub-errors to the end-user.

Reply:
Okay, let's chat about it tomorrow. I don't remember this point specifically, might be forgetting.